spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/SparkCatalog.java

```diff
@@ -361,32 +359,18 @@ public boolean purgeTable(Identifier ident) {
       ValidationException.check(
           PropertyUtil.propertyAsBoolean(table.properties(), GC_ENABLED, GC_ENABLED_DEFAULT),
           "Cannot purge table: GC is disabled (deleting files may corrupt other tables)");
-      String metadataFileLocation =
-          ((HasTableOperations) table).operations().current().metadataFileLocation();
 
-      boolean dropped = dropTableWithoutPurging(ident);
-
-      if (dropped) {
-        // check whether the metadata file exists because HadoopCatalog/HadoopTables
-        // will drop the warehouse directly and ignore the `purge` argument
-        boolean metadataFileExists = table.io().newInputFile(metadataFileLocation).exists();
-
-        if (metadataFileExists) {
-          SparkActions.get().deleteReachableFiles(metadataFileLocation).io(table.io()).execute();
```

Comment:
for some historical context: this was mainly introduced because `CatalogUtil.dropTableData` (which is being called by catalogs during purging) was too slow on giant tables